src/content/app/tools/blast/components/blast-settings/BlastSettings.tsx

```diff
@@ -0,0 +1,357 @@
+              selectedOption: blastParameters.match_scores as string,
+              onChange: (value: string) =>
+                onBlastParameterChange('match_scores', value)
+            })}
+          {buildSelect({
+            ...(getParameterData('wordsize') as BlastSelectSetting),
+            selectedOption: blastParameters.match_scores as string,
+            onChange: (value: string) =>
+              onBlastParameterChange('wordsize', value)
+          })}
+          {buildSelect({
+            ...(getParameterData('gapopen') as BlastSelectSetting),
+            selectedOption: blastParameters.gapopen as string,
+            onChange: (value: string) =>
+              onBlastParameterChange('gapopen', value)
+          })}
+          {buildSelect({
+            ...(getParameterData('gapext') as BlastSelectSetting),
+            selectedOption: blastParameters.gapopen as string,
+            onChange: (value: string) => onBlastParameterChange('gapext', value)
+          })}
+          {databaseSequenceType === 'protein' &&
+            buildSelect({
+              ...(getParameterData('matrix') as BlastSelectSetting),
+              selectedOption: blastParameters.gapopen as string,
+              onChange: (value: string) =>
+                onBlastParameterChange('matrix', value)
+            })}
+        </div>
+      )}
+    </>
+  );
+};
+
+const SequenceSwitcher = (props: {
+  sequenceType: string;
+  onChange: (sequenceType: string) => void;
+}) => {
+  const options = [
+    {
+      label: 'Nucleotide',
+      value: 'dna'
+    },
+    {
+      label: 'Protein',
+      value: 'protein'
+    }
+  ];
+  return (
+    <RadioGroup
+      options={options}
+      selectedOption={props.sequenceType}
+      onChange={(val) => props.onChange(val as string)}
+    />
+  );
+};
+
+// to be replaced with a simple select component
+const buildSelect = (setting: {
+  options: { label: string; value: string }[];
+  label: string;
+  selectedOption: string;
+  onChange: (value: string) => void;
+}) => {
+  const options = setting.options.map((option) => (
+    <option key={option.value} value={option.value}>
+      {option.label}
+    </option>
+  ));
+
+  const onChange = (e: FormEvent<HTMLSelectElement>) => {
+    const value = e.currentTarget.value;
+    setting.onChange(value);
+  };
+
+  return (
+    <div className={styles.select}>
+      <label>
+        <span>{setting.label}</span>
+        <select value={setting.selectedOption} onInput={onChange}>
+          {options}
+        </select>
+      </label>
+    </div>
+  );
+};
+
+const buildCheckbox = (params: {
+  label: string;
+  options: {
+    true: string;
+    false: string;
+  };
+  selectedOption: string;
+  onChange: (value: string) => void;
+}) => {
+  const { label, options, selectedOption, onChange } = params;
+  const isChecked = options.true === selectedOption;
+
+  const onCheckboxChange = (isChecked: boolean) => {
+    onChange(options[`${isChecked}`]);
```

Comment:
It is slightly confusing to read `isChecked` defined with same name one after the other. Should we rename one of it?